windows/win_service.ps1

```diff
@@ -45,7 +48,29 @@ If ($startMode) {
 $svcName = $name
 $svc = Get-Service -Name $svcName -ErrorAction SilentlyContinue
 If (-not $svc) {
-    Fail-Json $result "Service '$svcName' not installed"
+    If ($state -eq "present") {
+        If (-not $path) {
+            Fail-Json $result "path must be specified when state is '$state'"
+        }
+        New-Service -Name $svcName -DisplayName $display_name -BinaryPathName $path -Description $description
+        $svc = Get-Service -Name $svcName -ErrorAction SilentlyContinue
+        
+        Set-Attr $result "changed" $true
+        Set-Attr $result "state" $state
+    } ElseIf ($state -eq "absent") {
+	Exit-Json $result
+    } Else {
+	Fail-Json $result "Service '$svcName' not installed"
+    }
+} Else {
+    If ($state -eq "absent") {
+        Stop-Service -Name $svcName
```

Comment:
Would it be possible to add some error handling here?
If the stop fails this code will attempt to delete the service and I think you are likely to get 'service marked for deletion' state if this happens.  In my experience the only way to clear a 'service marked for deletion' is to reboot the windows machine.